### ISSUE TITLE
The values length for key 'data source' exceeds its limit

### ISSUE THREAD

**Issue by DGWright61** [dataplat/dbatools#8714]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

```powershell
@{writeErrorStream=True; PSMessageDetails=; Exception=System.Exception: The value's length for key 'data source' exceeds it's limit of '128'. ---> System.Exception: Error connecting to [myserver]: The value's length for key 'data source' exceeds it's limit of '128'. ---> System.Management.Automation.MethodInvocationException: Exception calling "ExecuteWithResults" with "1" argument(s): "Failed to connect to server Data Source=myserver;Integrated Security=True;MultipleActiveResultSets=False;Encrypt=False;TrustServerCertificate=False;Packet Size=4096;Application Name="dbatools PowerShell module - dbatools.io"." ---> Microsoft.SqlServer.Management.Common.ConnectionFailureException: Failed to connect to server Data Source=myserver;Integrated Security=True;MultipleActiveResultSets=False;Encrypt=False;TrustServerCertificate=False;Packet Size=4096;Application Name="dbatools PowerShell module - dbatools.io". ---> System.ArgumentException: The value's length for key 'data source' exceeds it's limit of '128'.
   at System.Data.SqlClient.SqlConnectionString..ctor(String connectionString)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnectionOptions(String connectionString, DbConnectionOptions previous)
   at System.Data.ProviderBase.DbConnectionFactory.GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, DbConnectionOptions& userConnectionOptions)
   at System.Data.SqlClient.SqlConnection.ConnectionString_Set(DbConnectionPoolKey key)
   at System.Data.SqlClient.SqlConnection.set_ConnectionString(String value)
   at Microsoft.SqlServer.Management.Common.ConnectionManager.get_SqlConnectionObject()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InternalConnect(WindowsIdentity impersonatedIdentity)
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
   --- End of inner exception stack trace ---
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.PoolConnect()
   at Microsoft.SqlServer.Management.Common.ServerConnection.ExecuteWithResults(String sqlCommand, Boolean retry)
   at CallSite.Target(Closure , CallSite , Object , String )
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---; TargetObject=myserver; CategoryInfo=ConnectionError: (myserver:PSObject) [Write-Error], Exception; FullyQualifiedErrorId=dbatools_Get-DbaDbLogShipError,Stop-Function; ErrorDetails=The value's length for key 'data source' exceeds it's limit of '128'.; InvocationInfo=System.Management.Automation.InvocationInfo; ScriptStackTrace=at Stop-Function, C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.143\allcommands.ps1: line 149040
at Get-DbaDbLogShipError<Process>, C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.143\allcommands.ps1: line 39175
at <ScriptBlock>, <No file>: line 1; PipelineIterationInfo=System.Collections.ObjectModel.ReadOnlyCollection`1[System.Int32]}
```

### Steps to Reproduce

```powershell
PS> Get-DbaDbLogShipError -SqlInstance myserver -Database MyDB
```

### Please confirm that you are running the most recent version of dbatools

1.1.143

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

Name                           Value                                                                                                  
----                           -----                                                                                                  
PSVersion                      5.1.19041.2364                                                                                         
PSEdition                      Desktop                                                                                                
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                
BuildVersion                   10.0.19041.2364                                                                                        
CLRVersion                     4.0.30319.42000                                                                                        
WSManStackVersion              3.0                                                                                                    
PSRemotingProtocolVersion      2.3                                                                                                    
SerializationVersion           1.1.0.1                                                                                                

### SQL Server Edition and Build number

SQL Server 2016 Standard Edition

### .NET Framework Version

PSChildName                      Version
-----------                      -------
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926
Windows Presentation Foundation  3.0.6920.4902
v3.5                             3.5.30729.4926
Client                           4.8.04084
Full                             4.8.04084
Client                           4.0.0.0

**Comment by wsmelton** [comment]
Please grab the T-SQL statement used in the command [Get-DbaLogShipError](https://github.com/dataplat/dbatools/blob/development/functions/Get-DbaDbLogShipError.ps1) and provide the output from your instance.

**Comment by TrondSO** [comment]
Hi I'm no expret but I did have simler problem. I did discover  that I did not have all the SQL Assamly's loaded:  If you look at my post: Powershell and dbatools - Get-DbaAvailabilityGroup
hmm, could not find any way to like my post, proberly because I'm a newbee :). (sorry for the spam)

So here are what i did find out:

I did only load the following Assembly's:
"Microsoft.SqlServer.Management.Common",
"Microsoft.SqlServer.Smo",

When i did load the rest of them:

"Microsoft.SqlServer.Dmf ",
"Microsoft.SqlServer.Instapi ",
"Microsoft.SqlServer.SqlWmiManagement ",
"Microsoft.SqlServer.ConnectionInfo ",
"Microsoft.SqlServer.SmoExtended ",
"Microsoft.SqlServer.SqlTDiagM ",
"Microsoft.SqlServer.SString ",
"Microsoft.SqlServer.Management.RegisteredServers ",
"Microsoft.SqlServer.Management.Sdk.Sfc ",
"Microsoft.SqlServer.SqlEnum ",
"Microsoft.SqlServer.RegSvrEnum ",
"Microsoft.SqlServer.WmiEnum ",
"Microsoft.SqlServer.ServiceBrokerEnum ",
"Microsoft.SqlServer.ConnectionInfoExtended ",
"Microsoft.SqlServer.Management.Collector ",
"Microsoft.SqlServer.Management.CollectorEnum",
"Microsoft.SqlServer.Management.Dac",
"Microsoft.SqlServer.Management.DacEnum",
"Microsoft.SqlServer.Management.Utility"

There was no problem, the Get-DbaAvailabilityGroup worked like a charme :)
Thank you so much for the replay here.
Here are the links that I did use:
https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2008-r2/cc281962(v=sql.105)?redirectedfrom=MSDN
https://learn.microsoft.com/en-us/sql/powershell/load-the-smo-assemblies-in-windows-powershell?view=sql-server-ver16

Hope this can help othere that get's this problem :).
